### PR TITLE
fix: fix torch.hub.load warning and tensor/numpy bug in deit.py

### LINKED ISSUE #485: imagenet/multiedge_inference_bench: save_weights() stores PyTorch tensors instead of numpy arrays in deit.py
**What happened**:
In deit.py, the save_weights() function in 
DeiTModelShard stores PyTorch tensor objects 
directly into a dictionary and then calls 
np.savez() to save them. But np.savez() only 
supports numpy arrays, not PyTorch tensors. 
This crashes with a TypeError when anyone tries 
to save model weights.

**What you expected to happen**:
The model weights should be saved successfully 
as a .npz file without any errors.

**How to reproduce it**:
1. Go to examples/imagenet/multiedge_inference_bench
2. Run DeiTModelShard.save_weights()
3. The program crashes with TypeError

**Anything else we need to know?**:
Fix: Change the line to:
weights[key] = val.detach().cpu().numpy()

### LINKED ISSUE #486: imagenet/multiedge_inference_bench: torch.hub.load() pretrained=True is deprecated in deit.py
**What happened**:
In deit.py, DeiTModelShard.save_weights() uses:
torch.hub.load(hub_repo, hub_model_name, pretrained=True)

The pretrained=True argument was deprecated in 
newer PyTorch versions. On modern PyTorch it raises 
FutureWarning and in very new versions it raises 
an error. This completely breaks the weight 
downloading process.

**What you expected to happen**:
The DeiT model weights should download and load 
successfully without any warnings or errors.

**How to reproduce it**:
1. Install latest PyTorch
2. Call DeiTModelShard.save_weights()
3. FutureWarning or TypeError is raised

**Anything else we need to know?**:
Fix: Change to:
model = torch.hub.load(
    hub_repo,
    hub_model_name,
    weights='DEFAULT'
)

### PR DESCRIPTION
**Pull request by rajdhruvsingh**
/kind bug

## What this PR does

Fixes two issues in `DeiTModelShard.save_weights()`:

1. Suppresses the `FutureWarning` raised by `torch.hub.load(..., pretrained=True)` in newer PyTorch versions.
2. Converts model weights to NumPy arrays before saving them with `np.savez()`.

### Issue #486

`torch.hub.load(..., pretrained=True)` emits a `FutureWarning` in newer PyTorch versions, which can clutter logs and affect compatibility. The warning is now suppressed during model loading.

### Issue #485

The original implementation stored PyTorch tensors directly in the `weights` dictionary before calling `np.savez()`. This can lead to serialization issues because `np.savez()` is intended to store NumPy arrays.

Converting tensors using:

```python
val.detach().cpu().numpy()
```

ensures the weights are saved in a NumPy-compatible format.

## Changes

* Wrapped `torch.hub.load()` in a `warnings.catch_warnings()` block and suppressed `FutureWarning`.
* Converted tensors to NumPy arrays before storing them in the weights dictionary.
* Updated:
  `examples/imagenet/multiedge_inference_bench/testalgorithms/automatic/models/transformers/deit.py`

## Which issue(s) this PR fixes

Fixes #485

Fixes #486

### CODE REVIEW

**Comment by rajdhruvsingh** [comment]
Thank you for the review from gemini-code-assist.

After investigating further, I understand the issue now.
`weights='DEFAULT'` applies to torchvision.models but
facebookresearch/deit's hubconf.py still expects `pretrained=True`
as its own parameter. Changing it would cause a TypeError.

I am investigating the correct approach and will update
the PR. The real fix likely involves suppressing the
FutureWarning while keeping pretrained=True, since we
cannot change Facebook's hub interface.

**Comment by rajdhruvsingh** [comment]
Note: I have several related PRs open that fix bugs 
in the same files (bert.py and deit.py), all stemming 
from the bug report in issue #491. 

Suggested merge order to avoid conflicts:
- bert.py PRs: #478 first, then #479, then #480
- deit.py PRs: #501 first, then #487

Happy to rebase any PR on top of another if needed. 
Please let me know the preferred approach.

**Review by MooreZheng — COMMENTED** [review]
/lgtm

**Review by MooreZheng — APPROVED** [review]
/approve
